src/3d/framegraph/qgsframegraph.cpp

```diff
@@ -424,13 +398,16 @@ void QgsFrameGraph::unregisterRenderView( const QString &name )
   }
 }
 
-bool QgsFrameGraph::registerRenderView( std::unique_ptr<QgsAbstractRenderView> renderView, const QString &name )
+bool QgsFrameGraph::registerRenderView( std::unique_ptr<QgsAbstractRenderView> renderView, const QString &name, Qt3DRender::QFrameGraphNode *topNode )
 {
   bool out;
   if ( mRenderViewMap.find( name ) == mRenderViewMap.end() )
   {
     mRenderViewMap[name] = std::move( renderView );
-    mRenderViewMap[name]->topGraphNode()->setParent( mMainViewPort );
+    if ( topNode )
+      mRenderViewMap[name]->topGraphNode()->setParent( topNode );
+    else
+      mRenderViewMap[name]->topGraphNode()->setParent( mMainViewPort );
```

Comment:
A matter of taste, but I find this easier to read:
```suggestion
    mRenderViewMap[name]->topGraphNode()->setParent( topNode ? topNode : mMainViewPort );
```

Reply:
fixed